logstash-core/src/main/java/org/logstash/instrument/metrics/timer/ConcurrentLiveTimerMetric.java

```diff
@@ -43,10 +48,13 @@ protected ConcurrentLiveTimerMetric(final String name) {
     @Override
     public <T, E extends Throwable> T time(ExceptionalSupplier<T, E> exceptionalSupplier) throws E {
         try {
-            trackedMillisState.getAndUpdate(TrackedMillisState::withIncrementedConcurrency);
+            trackedMillisState.getAndUpdate(existing -> existing.withIncrementedConcurrency(nanoTimeSupplier.getAsLong()));
```

Comment:
Note: the `nanoTimeSupplier.getAsLong()` for recording the start time still occurs inside the `UnaryConsumer` because if this thread loses and is retried, we don't want to record an old start time.